mlir/lib/Dialect/LLVMIR/IR/LLVMDialect.cpp

```diff
@@ -2319,6 +2319,23 @@ ParseResult LLVMFuncOp::parse(OpAsmParser &parser, OperationState &result) {
     result.addAttribute(getComdatAttrName(result.name), comdat);
   }
 
+  // Parse the optional frame_pointer
+  if (succeeded(parser.parseOptionalKeyword("frame_pointer"))) {
+    std::string string;
+    
+    if (parser.parseEqual() || parser.parseString(&string))
+      return failure();
+    if (!LLVM::symbolizeFramePointerKind(string))
+    { 
+      llvm::outs() << "failure: frame-pointer option not recognized: " << string << "\n";
+      return failure();
```

Comment:
Emitting errors in parsing code should be done via the `emitError` method of `parser` as this utilizes MLIRs diagnostics infrastructure. Writing to `llvm::outs` or any other global stream in particularly is heavly discouraged.